packages/queued-request-controller/src/QueuedRequestController.ts

```diff
@@ -251,18 +259,25 @@ export class QueuedRequestController extends BaseController<
    */
   async #processNextBatch() {
     const firstRequest = this.#requestQueue.shift() as QueuedRequest;
+    this.#networkClientIdOfCurrentBatch = firstRequest.networkClientId;
```

Comment:
nit, can we switch this line with the origin line right below so that originOfCurrentBatch assignment comes first? (Only because that's the ordering in the rest of the controller)

Reply:
Sure thing!

Reply:
done here: https://github.com/MetaMask/core/pull/4718/commits/2603f2530bddf9b1301d37cad319fdd93e5b8a28